### DISCUSSION TITLE
TypeSafety for NextJS getServerSideProps with serverSideTranslations

### DISCUSSION THREAD

**Discussion by dachewster99** [i18next/next-i18next#2076] [Q&A] [answered]
I have an application where we are using serverSideTranslations, and currently we are doing `locale!`, is there a best practice you recommend as we are trying to avoid the use of `!` to force typescript to be happy? It appears to not allow for undefined, and an empty string does not appear to use the defaultLocale as a fallback. 

```
export const getServerSideProps: GetServerSideProps = async ({ locale }) => {
  return {
    props: {
      ...(await serverSideTranslations(locale!)),
    },
  };
};
```

**Comment by abdulrahimiliasu** [comment, marked as answer]
How about `locale ?? "en"` instead.

**Reply by dachewster99** [reply]
> How about `locale ?? "en"` instead.

I saw that from one of the examples and went with that. Just curious if there was a best practice, but for my purpose that with fine
